components/api/src/main/java/com/hotels/styx/api/Url.java

```diff
@@ -529,7 +536,8 @@ public Builder authority(String authority) {
          * @return this builder
          */
         public Builder path(String path) {
-            this.path = path;
+            this.rawPath = path;
+            this.path = QueryStringDecoder.decodeComponent(path);
```

Comment:
We don't need to store the rawPath (or path). Only one of them is necessary. We should store the path component as-is regardless of encoding. As long as we take the raw path from *url()* method like so:

        public static Builder url(String value) {
            URI uri = URI.create(value);
            return new Builder()
                    .scheme(uri.getScheme())
                    .authority(uri.getAuthority())
                    .path(uri.getRawPath())               <--- here
                    .rawQuery(uri.getRawQuery())
                    .fragment(uri.getFragment());
        }

Reply:
Please change this accordingly. I have write some justification in https://github.com/HotelsDotCom/styx/pull/15#issuecomment-342511138.